### ISSUE TITLE
Error while compiling (Detect_MS_HyperV_AES)

### ISSUE THREAD

**Issue by gunph1ld** [veracrypt/VeraCrypt#83]
Hello, guys. I'm trying to compile from the sources and get an error:

```
$ make
Compiling cpu.c
../Crypto/cpu.c: In function ‘Detect_MS_HyperV_AES’:
../Crypto/cpu.c:214:3: error: ‘__try’ undeclared (first use in this function)
   __try
   ^~~~~
../Crypto/cpu.c:214:3: note: each undeclared identifier is reported only once for each function it appears in
../Crypto/cpu.c:215:3: error: expected ‘;’ before ‘{’ token
   {
   ^
/home/jack/git/VeraCrypt/src/Build/Include/Makefile.inc:20: recipe for target '../Crypto/cpu.o' failed
make[1]: *** [../Crypto/cpu.o] Error 1
Makefile:298: recipe for target 'all' failed
make: *** [all] Error 2
```
Fedora 24 4.5.5-300.fc24.x86_64

**Comment by frank-dittrich** [comment]
Checkout commit 2dbf366, this one still works.
The most recent changes for Windows apparently broke non-Windows builds.

On Fedora 24, I had to
```
dnf install wxGTK3-devel
```
and
```
ln -s /usr/bin/wx-config-3.0 /usr/bin/wx-config
```
to get a working build after a `git checkout 2dbf366`.
I just get tons of warnings ` warning: 'template<class> class std::auto_ptr' is deprecated [-Wdeprecated-declarations]`.

```
((HEAD detached at 2dbf366))src $ LC_ALL=C make
Compiling Buffer.cpp
Compiling Exception.cpp
Compiling Event.cpp
Compiling FileCommon.cpp
Compiling MemoryStream.cpp
Compiling Memory.cpp
Compiling PlatformTest.cpp
Compiling Serializable.cpp
Compiling Serializer.cpp
Compiling SerializerFactory.cpp
Compiling StringConverter.cpp
Compiling TextReader.cpp
Compiling Directory.cpp
Compiling File.cpp
Compiling FilesystemPath.cpp
Compiling Mutex.cpp
Compiling Pipe.cpp
Compiling Poller.cpp
Compiling Process.cpp
Unix/Process.cpp: In static member function 'static std::__cxx11::string VeraCrypt::Process::Execute(const string&, const std::__cxx11::list<std::__cxx11::basic_string<char> >&, int, VeraCrypt::ProcessExecFunctor*, const VeraCrypt::Buffer*)':
Unix/Process.cpp:193:4: warning: 'template<class> class std::auto_ptr' is deprecated [-Wdeprecated-declarations]
    auto_ptr <Serializable> deserializedObject;
    ^~~~~~~~
In file included from /usr/include/c++/6.1.1/memory:81:0,
                 from /home/fd/git/VeraCrypt/src/Platform/PlatformBase.h:19,
                 from Unix/Process.h:16,
                 from Unix/Process.cpp:18:
/usr/include/c++/6.1.1/bits/unique_ptr.h:49:28: note: declared here
   template<typename> class auto_ptr;
                            ^~~~~~~~
Compiling SyncEvent.cpp
Compiling SystemException.cpp
Compiling SystemInfo.cpp
Compiling SystemLog.cpp
Compiling Thread.cpp
Compiling Time.cpp
Updating library Platform.a
Compiling Cipher.cpp
Compiling EncryptionAlgorithm.cpp
Compiling EncryptionMode.cpp
In file included from EncryptionMode.cpp:15:0:
EncryptionThreadPool.h:47:4: warning: 'template<class> class std::auto_ptr' is deprecated [-Wdeprecated-declarations]
    auto_ptr <Exception> ItemException;
    ^~~~~~~~
In file included from /usr/include/c++/6.1.1/memory:81:0,
                 from /home/fd/git/VeraCrypt/src/Platform/PlatformBase.h:19,
                 from /home/fd/git/VeraCrypt/src/Platform/Platform.h:16,
                 from EncryptionMode.h:16,
                 from EncryptionMode.cpp:13:
/usr/include/c++/6.1.1/bits/unique_ptr.h:49:28: note: declared here
   template<typename> class auto_ptr;
                            ^~~~~~~~

[...]
```

**Comment by idrassi** [comment]
I have pushed a commit the fixes this issue: https://github.com/veracrypt/VeraCrypt/commit/3fb2eedab8ef586cd2686efba0b668a5070fd0af
The latest code from master should build without issues.

Indeed, there are many auto_ptr warnings but these are harmless and they will be dealt with later.
